driver-core/src/test/functional/com/mongodb/client/model/AggregatesFunctionalSpecification.groovy

```diff
@@ -380,6 +381,104 @@ class AggregatesFunctionalSpecification extends OperationFunctionalSpecification
+        def fromHelper = getCollectionHelper(new MongoNamespace(getDatabaseName(), fromCollectionName))
+
+        fromHelper.drop()
+
+        fromHelper.insertDocuments(
+            Document.parse('''{ _id: 0, name: "Bob Smith", friends: ["Anna Jones", "Chris Green"],
+                hobbies : ["tennis", "unicycling", "golf"] }'''),
+            Document.parse('''{ _id: 1, name: "Anna Jones", friends: ["Bob Smith", "Chris Green", "Joe Lee"],
+                 hobbies : ["archery", "golf", "woodworking"] }'''),
+            Document.parse('''{ _id: 2, name: "Chris Green", friends: ["Anna Jones", "Bob Smith"],
+                hobbies : ["knitting", "frisbee"] }'''),
+            Document.parse('''{ _id: 3, name: "Joe Lee", friends: ["Anna Jones", "Fred Brown"],
+                hobbies : [ "tennis", "golf", "topiary" ] }'''),
+            Document.parse('''{ _id: 4, name: "Fred Brown", friends: ["Joe Lee"],
+                hobbies : [ "travel", "ceramics", "golf" ] }'''))
+
+
+        def lookupDoc = graphLookup('contacts', new BsonString('$friends'), 'friends', 'name', 'golfers',
+                new GraphLookupOptions()
+                        .restrictSearchWithMatch(eq('hobbies', 'golf')))
+
+        when:
+        def results = fromHelper.aggregate([lookupDoc,
+                                            unwind('$golfers'),
+                                            sort(new Document('_id', 1)
+                                                    .append('golfers._id', 1))])
+
+        then:
+        results.subList(0, 4) == [
+                Document.parse('''{ _id: 0, name: "Bob Smith", friends: ["Anna Jones", "Chris Green"],
+                        hobbies : ["tennis", "unicycling", "golf"], golfers: {_id: 0, name: "Bob Smith",
+                        friends: ["Anna Jones", "Chris Green"], hobbies : ["tennis", "unicycling", "golf"] } }'''),
+                Document.parse('''{ _id: 0, name: "Bob Smith", friends: ["Anna Jones", "Chris Green"],
+                       hobbies: ["tennis", "unicycling", "golf"], golfers:{ _id: 1, name: "Anna Jones",
+                       friends: ["Bob Smith", "Chris Green", "Joe Lee"], hobbies : ["archery", "golf", "woodworking"] } } }'''),
+                Document.parse('''{ _id: 0, name: "Bob Smith", friends: ["Anna Jones", "Chris Green"],
+                       hobbies: ["tennis", "unicycling", "golf"], golfers: { _id: 3, name: "Joe Lee",
+                       friends: ["Anna Jones", "Fred Brown"], hobbies : [ "tennis", "golf", "topiary" ] } }'''),
+                Document.parse('''{ _id: 0, name: "Bob Smith", friends: ["Anna Jones", "Chris Green"],
+                       hobbies: ["tennis", "unicycling", "golf"], golfers:{ _id: 4, name: "Fred Brown", friends: ["Joe Lee"],
+                       hobbies : [ "travel", "ceramics", "golf" ] } }''')
+        ]
+
+        cleanup:
+        fromHelper?.drop()
+    }
+
+    @IgnoreIf({ !serverVersionAtLeast(3, 4) })
```

Comment:
This test seems redundant.  Is this testing anything that's not already being tested above?

Reply:
Yes, it's redundant. I have added 2 tests : 

- with only  restricSearchWithMatch  option : 
```
new GraphLookupOptions().restrictSearchWithMatch(eq('hobbies', 'golf')))
```

- with all options ;
``` 
new GraphLookupOptions().depthField('depth').maxDepth(0).restrictSearchWithMatch(eq('hobbies', 'golf')))
```

I can remove one tetst. Wich ?

Reply:
I'd remove the second one with all options

Reply:
done !